web3_proxy/src/admin_queries.rs

```diff
@@ -0,0 +1,134 @@
+use crate::app::Web3ProxyApp;
+use crate::frontend::errors::FrontendErrorResponse;
+use crate::user_queries::get_user_id_from_params;
+use anyhow::Context;
+use axum::{
+    Json,
+    headers::{authorization::Bearer, Authorization},
+    TypedHeader,
+};
+use axum::response::{IntoResponse, Response};
+use entities::{admin, user, user_tier};
+use ethers::prelude::Address;
+use hashbrown::HashMap;
+use http::StatusCode;
+use migration::sea_orm::{self, ActiveModelTrait, ColumnTrait, EntityTrait, IntoActiveModel, QueryFilter};
+use log::info;
+use redis_rate_limiter::redis::AsyncCommands;
+use crate::frontend::errors::FrontendErrorResponse::AccessDenied;
+
+// TODO: Add some logic to check if the operating user is an admin
+// If he is, return true
+// If he is not, return false
+// This function is used to give permission to certain users
+
+
+pub async fn query_admin_modify_usertier<'a>(
+    app: &'a Web3ProxyApp,
+    bearer: Option<TypedHeader<Authorization<Bearer>>>,
+    params: &'a HashMap<String, String>
+) -> Result<Response, FrontendErrorResponse> {
+
+    // Quickly return if any of the input tokens are bad
+    let user_address: Vec<u8> = params
+        .get("user_address")
+        .ok_or_else(||
+            FrontendErrorResponse::StatusCode(
+                StatusCode::BAD_REQUEST,
+                "Unable to find user_address key in request".to_string(),
+                None,
+            )
+        )?
+        .parse::<Address>()
+        .map_err(|err| {
+            FrontendErrorResponse::StatusCode(
+                StatusCode::BAD_REQUEST,
+                "Unable to parse user_address as an Address".to_string(),
+                Some(err.into()),
+            )
+        })?
+        .to_fixed_bytes().into();
+    let user_tier_title = params
+        .get("user_tier_title")
+        .ok_or_else(|| FrontendErrorResponse::StatusCode(
+            StatusCode::BAD_REQUEST,
+            "Unable to get the user_tier_title key from the request".to_string(),
+            None,
+        ))?;
+
+    // Prepare output body
+    let mut response_body = HashMap::new();
+
+    // Establish connections
+    let db_conn = app.db_conn().context("query_admin_modify_user needs a db")?;
+    let db_replica = app
```

Comment:
Since this endpoint writes to the database, do not get `db_replica`. Just use db_conn for all queries

Reply:
I replaced the majority with read-replicas now. The reason is that there is a case where the user-tier needs not to be modified, in this case writing is unnecessary. I can still go ahead and just use db_conn though if you prefer

Reply:
I think that's safest. I've had races in the past cause problems

Reply:
`get_user_id_from_params` seems to still be using it, i will not delete it, but replace usage in all of the file (except for this function) with db_conn